tests/model_registry/model_registry/python_client/signing/utils.py

```
@@ -59,3 +68,67 @@ def create_connection_type_field(
+        response.raise_for_status()  # Raise exception for HTTP errors
+
+        # Calculate SHA256 checksum
+        checksum = hashlib.sha256(response.content).hexdigest()
+        LOGGER.info(f"Calculated root.json checksum: {checksum}")
+
+    except requests.RequestException as e:
+        LOGGER.error(f"Failed to download root.json from {sigstore_tuf_url}: {e}")
+        raise
+    except (ValueError, OSError) as e:
+        LOGGER.error(f"Failed to calculate checksum: {e}")
+        raise RuntimeError(f"Checksum calculation failed: {e}")
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)